src/main/java/io/javalin/core/JavalinServlet.kt

```diff
@@ -74,7 +75,44 @@ class JavalinServlet(
                 }
             }
             if (endpointEntries.isEmpty() && type != HandlerType.HEAD && type != HandlerType.GET) {
```

Comment:
I did not study the code thoroughly, but I think it would be better to throw HaltException(405) (if possible) here instead of rendering 405 as if it would be normal. That is because I think that exception mapping won't be triggered for this https://javalin.io/documentation#exception-mapping

Here I mentioned that there might be couple of places in Javalin that are hardcoded, which may be undersirable given some strict requirements, this would be one of them.
https://github.com/tipsy/javalin/issues/219